plugins/woocommerce/src/StoreApi/Routes/V1/Agentic/CheckoutSessionsUpdate.php

```diff
@@ -0,0 +1,245 @@
+				array( 'status' => 404 )
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Use the checkout_session_id as Cart-Token, and set the respective values to HTTP header and request.
+	 *
+	 * @param \WP_REST_Request $request Request object.
+	 * @return bool|null
+	 */
+	protected function has_cart_token( \WP_REST_Request $request ) {
+		$session_id = $request->get_param( 'checkout_session_id' );
+		if ( is_null( $this->has_cart_token ) ) {
+			$this->has_cart_token = CartTokenUtils::validate_cart_token( $session_id );
+		}
+
+		// This allows the session will be loaded later without any further intervention.
+		if ( true === $this->has_cart_token ) {
+			$request->set_header( 'Cart-Token', $session_id );
+			$_SERVER['HTTP_CART_TOKEN'] = $session_id;
+		}
+
+		return $this->has_cart_token;
+	}
+
+	/**
+	 * Handle the request and return a valid response for this endpoint.
+	 *
+	 * @param \WP_REST_Request $request Request object.
+	 * @return \WP_REST_Response|\WP_Error
+	 */
+	protected function get_route_post_response( \WP_REST_Request $request ) {
+		// Update items if provided.
+		$items = $request->get_param( 'items' );
+		if ( null !== $items ) {
+			// Clear existing cart items and replace with new ones.
+			WC()->cart->empty_cart();
+
+			$error = AgenticCheckoutUtils::add_items_to_cart( $items, $this->cart_controller );
+			if ( $error ) {
+				return $error;
+			}
+		}
+
+		// Update buyer information if provided.
+		$buyer = $request->get_param( 'buyer' );
+		if ( null !== $buyer ) {
+			AgenticCheckoutUtils::set_buyer_data( $buyer, WC()->customer );
+		}
+
+		// Update fulfillment address if provided.
+		$address = $request->get_param( 'fulfillment_address' );
+		if ( null !== $address ) {
+			AgenticCheckoutUtils::set_fulfillment_address( $address, WC()->customer );
+		}
+
+		// Update selected shipping method if provided.
+		$fulfillment_option_id = $request->get_param( 'fulfillment_option_id' );
+		if ( null !== $fulfillment_option_id ) {
+			$option_id = wc_clean( (string) $fulfillment_option_id );
+			$packages  = WC()->shipping()->get_packages();
+			foreach ( $packages as $package ) {
+				foreach ( (array) ( $package['rates'] ?? array() ) as $rate ) {
+					if ( $rate->get_id() === $option_id ) {
+						WC()->session->set( SessionKey::CHOSEN_SHIPPING_METHODS, array( $option_id ) );
+						break 2;
+					}
+				}
+			}
+		}
+
+		// Calculate totals after all updates.
+		WC()->cart->calculate_totals();
+
+		// Build response from canonical cart schema.
+		$response = rest_ensure_response( $this->schema->get_item_response( WC()->cart ) );
+
+		// Add protocol headers.
+		return AgenticCheckoutUtils::add_protocol_headers( $response, $request );
+	}
+
+	/**
+	 * When the cart is updated, create or update draft order.
+	 * Overrides parent to ensure draft order is created if it doesn't exist.
+	 *
+	 * @param \WP_REST_Request $request Request object.
+	 */
+	protected function cart_updated( \WP_REST_Request $request ) {
```

Comment:
Since we're moving to use cart-token, which is mapped to the session and cart. 
Therefore, an order should be only created when AI triggers the payment, i.e. specs endpoint `/complete`. 

This is similar to the current StoreApi, see `Checkout::create_or_update_draft_order`. 


This simplifies a lot and we can keep relying on the parent method `AbstractCartRoute::cart_updated`, which is only the sync the order after a cart has been through `Checkout` but not finished yet. That also prevents a lot of unnecessary draft orders.